contributing.md

```diff
@@ -0,0 +1,81 @@
+Please use the Github issue board to track and take ownership of issues. 
+
+Please let us know if you encounter a bug by [creating an issue](https://github.com/opendifferentialprivacy/whitenoise-core/issues).
+We appreciate all contributions. We welcome pull requests with bug-fixes without prior discussion.
+
+If you plan to contribute new features, utility functions or extensions to the core, please first open an issue and 
+discuss the feature with us. Sending a PR without discussion might end up resulting in a rejected PR, because we might
+ be taking the core in a different direction than you might be aware of.
+
+For a description of the library architecture and installation instructions, see whitenoise-core/README.md. 
+Before contributing, we recommend following the Getting Started guide in that README to get sample 
+notebooks running and to gain some familiarity with the library structure.
+
+General Contribution Guidelines
+===============================
+
+- Use the issues boards
+- Anything implemented from the differential privacy literature should contain clear citations and/or whitepapers 
+explaining any necessary derivations.
+- Code should be clearly documented and include testing. See https://doc.rust-lang.org/rustdoc/documentation-tests.html
+for an explanation on how to implement block tests and https://doc.rust-lang.org/book/ch11-03-test-organization.html for
+information on testing modules.
+
+Contributing to the Validator
+=============================
+
+Components need to be added to both the protobuf and to the source codebase. 
+
+### 1. Add component to protobuf in validator-rust/prototypes:
+
+Add a json file to the components subdirectory. An entry in components.proto will be made automatically.
+### 2. Add component to src/components:
+
+This will look different depending on the type of component you are contributing. The following describes the traits
+that must be implemented for common possible types of components. 
+
+### For new statistics:
+  - A `Component` implementation containing `propagate_property` function describing 
+        how to propagate input data through that node in the graph.
+  - A `Sensitivity` implementation with `compute_sensitivity` function that describes how to compute
+        the statistic for all combinations of implemented privacy definitions and sensitivity spaces
+        - any derivations used for sensitivities should be derived in the `whitepapers/sensitivities`,
+        or there should be a citation to published sensitivity analysis (or both).
+        - include comment in `Sensitivity` implementation that links to the location of the proof
+### For new DP statistics:
+  - An `Expandable` implementation with `expand_component` function which describes how to expand the computation graph
+        to insert each component of the differentially private calculation. (Note that not all the nodes of the
+        expanded graph will be releasable)
```

Comment:
Releasable is a property. That property can only be set by the mechanism. So only the nodes including and after the mechanism are releasable.

Reply:
I misread this. "not all". Sorry.